docs/user-guide/how_to_tune_volcano_performance.md

```diff
@@ -0,0 +1,571 @@
+
+![V-Shaped CREATE Time with Different Worker Threads](../images/performance-tuning/v-shaped-performance-curve-worker-threads.png)
+
+### 4.3 Hypothesis 3: Contention Between Controller and Scheduler
+
+We observed that the stair-step stalls for `CREATE` (performed by the Controller) and `SCHEDULE` (performed by the Scheduler) never occurred simultaneously. We also saw a large number (over 3000) of `409 Conflict` errors in the `kube-apiserver` logs. Further investigation revealed that **the root cause is contention from two independent components concurrently writing to a shared resource, the K8s API-Server/ETCD. These concurrent writes trigger optimistic lock conflicts**, leading to request retries and queuing, which manifests macroscopically as alternating execution.
+
+#### 4.3.1 Analysis of Independence at the Source Code Level
+
+A careful analysis of the Volcano source code shows that the `CREATE` functionality belongs to the Controller component, while the `SCHEDULE` functionality belongs to the Scheduler component. There are no direct calls, dependencies, or other relationships between them at the code level:
+
+```go
+// pkg/controllers/job/job_controller_actions.go
+func (cc *jobcontroller) syncJob(jobInfo *apis.JobInfo, updateStatus state.UpdateStatusFn) error {
+    // Controller is responsible for Pod creation
+    for _, pod := range podToCreateEachTask {
+        go func(pod *v1.Pod) {
+            defer waitCreationGroup.Done()
+            newPod, err := cc.kubeClient.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+            // ...
+        }(pod)
+    }
+    waitCreationGroup.Wait()
+}
+
+// pkg/scheduler/actions/allocate/allocate.go
+func (alloc *Action) Execute(ssn *framework.Session) {
+    // Scheduler is responsible for Pod scheduling
+    for _, task := range tasks {
+        // Scheduling decision and binding
+        ssn.Bind(task, node)
+    }
+}
+```
+
+#### 4.3.2 Analysis of Contention at the API-Server Level
+
+**The real contention occurs at the K8s API-Server level**. After completing their respective operations, both components must submit their results (creating a Pod / updating a Pod's status) to the **K8s API-Server**, which then handles writing to etcd. When both components operate on the same Pod object simultaneously, K8s's optimistic locking mechanism is triggered. One component's write request will fail with a `409 Conflict` error due to a `resourceVersion` mismatch, forcing a retry. The large volume of concurrent requests, conflicts, and retries ultimately manifests as an alternating execution pattern where one works while the other waits.
+
+**Request Flow Analysis**:
+
+```go
+// Controller's Pod creation request flow
+JobController.syncJob() 
+  → kubeClient.CoreV1().Pods().Create() 
+    → kube-apiserver 
+      → etcd write
+
+// Scheduler's Pod scheduling request flow
+VolcanoScheduler.allocate() 
```

Comment:
vc-controller create pod and and vc-scheduler update will not generate 409 conflicts here, only after the controller is created, the scheduler can watch the latest resourceVersion pod and try to schedule, what does this request flow intend to express?